Patterns/Spanish/Spanish-DateTime.yaml

```diff
@@ -394,7 +394,7 @@ NowRegex: !simpleRegex
 SuffixRegex: !simpleRegex
   def: ^\s*(((y|a|en|por)\s+la|al)\s+)?(mañana|madrugada|medio\s*d[ií]a|(?<!(m[áa]s\s+))tarde|noche)\b
 TimeOfDayRegex: !nestedRegex
-  def: \b(?<timeOfDay>(({LaterEarlyRegex}\s+)((del?|en|por)(\s+(el|los?|las?))?\s+)?|(en|por)\s+(el|l[ao]s?)\s+)?(mañana|madrugada|pasado\s+(el\s+)?medio\s?d[ií]a|(?<!((m[áa]s|tan)\s+))tarde|noche))\b
+  def: \b((?<timeOfDay>(({LaterEarlyRegex}\s+)((del?|en|por)(\s+(el|los?|las?))?\s+)?)?(mañana|madrugada|pasado\s+(el\s+)?medio\s?d[ií]a|(?<!((m[áa]s|tan)\s+))tarde|noche))|(en|por)\s+las?\s+mañana)\b
```

Comment:
Why is this causing a problem in Java? Also, now it's not fully correct, right? As cases like "por la madrugada" and others without later/early won't be captured anymore.

Reply:
I have not fully understood the reason for the failure, I can investigate the problem in more details if needed. 
Now the regex is in its original form, a part for the additional pattern "(en|por)\s+las?\s+mañana". 
For a case like "por la madrugada", it will extract only "madrugada". 
The additional pattern "por la mañana" is needed because isolated extractions of "mañana" are filtered out due to the ambiguity in the meaning "morning/tomorrow".